buildlib/travis-build

```diff
@@ -10,6 +10,7 @@ cd build
 # The goal is warning free compile on latest gcc.
 CC=gcc-6 CFLAGS=-Werror cmake -GNinja ..
 ninja
+../buildlib/check-build --src ..
```

Comment:
@jgunthorpe , i merged the code as is. it can be changed if needed in followup patch.Anyway it looks like the patches to build infrastructure are coming from two of us only.